test/Unit/Task/FileSizeTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHPTest\Unit\Task;
+
+use GrumPHP\Collection\FilesCollection;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use GrumPHP\Task\FileSize;
+use GrumPHP\Task\TaskInterface;
+use Prophecy\Prophecy\ObjectProphecy;
+use Symfony\Component\Finder\SplFileInfo;
+
+class FileSizeTest extends AbstractTaskTestCase
+{
+    protected function provideTask(): TaskInterface
+    {
+        return new FileSize();
+    }
+
+    public function provideConfigurableOptions(): iterable
+    {
+        yield 'defaults' => [
+            [],
+            [
+                'max_size' => '10M',
+                'ignore_patterns' => [],
+            ]
+        ];
+
+        yield 'invalidcase' => [
+            [
+                'ignore_patterns' => 'thisisnotanarray'
+            ],
+            null
+        ];
+    }
+
+    public function provideRunContexts(): iterable
+    {
+        yield 'run-context' => [
+            true,
+            $this->mockContext(RunContext::class)
+        ];
+
+        yield 'pre-commit-context' => [
+            true,
+            $this->mockContext(GitPreCommitContext::class)
+        ];
+
+        yield 'other' => [
+            false,
+            $this->mockContext()
+        ];
+    }
+
+    public function provideFailsOnStuff(): iterable
+    {
+        /** @var ContextInterface|ObjectProphecy $context */
+        $context = $this->prophesize(RunContext::class);
```

Comment:
I'dd create a specialized mockContext method for this class in this case.
You probably want to mock the file size instead of the collection.
That way you are more flexible in how it works and you will actually validate the task.